packages/components/inputs/select-utils/src/create-select-styles.ts

```diff
@@ -97,10 +97,15 @@ type TDesignTokenName = keyof typeof designTokens;
 const getHorizontalConstraintValue = (
   value?: TProps['minMenuWidth'] | TProps['maxMenuWidth']
 ) => {
-  return (
-    designTokens[`constraint${value}` as TDesignTokenName] ||
-    designTokens.constraint3
-  );
+  switch (value) {
+    case 'auto':
+      return 'initial';
+    default:
+      return (
+        designTokens[`constraint${value}` as TDesignTokenName] ||
+        designTokens.constraintScale
```

Comment:
From this helper point of view, I think it's safer to default to 100%.

Using this approach, even if we don't transform the incoming `scale` value, we will return that value anyway.